### ISSUE TITLE
Creating an Element Error: this.opts.filename.replce is not a function

### ISSUE THREAD

**Issue by Tykin** [Polymer/polymer-cli#206]
Trying to create a polymer element using the polymer client tool in windows. The creation step fails repeatedly. Using Node 4.4.0.

```

C:\Users\tykin\test\test-element>polymer init
? Which starter template would you like to use? polymer-init-element:app
info:    Running template element
? Element name test-element
? Brief description of the element None
error:   Uncaught exception: TypeError: this.opts.filename.replace is not a function
error:   TypeError: this.opts.filename.replace is not a function
    at Object.Template.generateSource (C:\Program Files\nodejs\node_modules\polymer-cli\node_modules\ejs\lib\ejs.js:528:68)
    at Object.Template.compile (C:\Program Files\nodejs\node_modules\polymer-cli\node_modules\ejs\lib\ejs.js:444:12)
    at Object.compile (C:\Program Files\nodejs\node_modules\polymer-cli\node_modules\ejs\lib\ejs.js:289:16)
    at handleCache (C:\Program Files\nodejs\node_modules\polymer-cli\node_modules\ejs\lib\ejs.js:148:16)
    at Object.exports.render (C:\Program Files\nodejs\node_modules\polymer-cli\node_modules\ejs\lib\ejs.js:316:10)
    at copy.process (C:\Program Files\nodejs\node_modules\polymer-cli\node_modules\mem-fs-editor\lib\actions\copy-tpl.js:14:18)
    at applyProcessingFunc (C:\Program Files\nodejs\node_modules\polymer-cli\node_modules\mem-fs-editor\lib\actions\copy.js:13:16)
    at EditionInterface.exports._copySingle (C:\Program Files\nodejs\node_modules\polymer-cli\node_modules\mem-fs-editor\lib\actions\copy.js:51:16)
    at EditionInterface.<anonymous> (C:\Program Files\nodejs\node_modules\polymer-cli\node_modules\mem-fs-editor\lib\actions\copy.js:38:10)
    at Array.forEach (native)
```

**Comment by yordis** [comment]
@Tykin Actually even for application type I get this message as well

**Comment by MarcosGalaviz** [comment]
I have the same issue :(

**Comment by FredKSchott** [comment]
I can reproduce this on Windows as well, not on Linux/Mac.

We're actively tracking and fixing a few different Windows bugs that have popped up since launching. See #199 for our current work to get tests passing on Windows. We'll tackle this as soon as possible once that's merged.

**Comment by dcoblentz** [comment]
I'm seeing this on mac as well.

**Comment by FredKSchott** [comment]
@dcoblentz Interesting, we've previously been tracking this as a Windows bug. What version of Node.js are you running?

**Comment by FredKSchott** [comment]
Okay, it's definitely starting to look like a Node.js v4.x issue. I can reproduce with node v4.4.4 on both Mac & Windows, but cannot reproduce with node v5.x or v6.x.

**Comment by dcoblentz** [comment]
I'm on v6.2.0. I had also tried in v5.5.0

**Comment by Tykin** [comment]
I can also reproduce in Mac. Using node v5.7.0 and npm 3.6.0.  I am going to update to the latest versions and try again.

```
Jerrys-MBP:test-element Tykin$ polymer init
? Which starter template would you like to use? element
info:    Running template element
? Element name test-element
? Brief description of the element Description
error:   Uncaught exception: TypeError: this.opts.filename.replace is not a function
error:   TypeError: this.opts.filename.replace is not a function
    at Object.Template.generateSource (/usr/local/lib/node_modules/polymer-cli/node_modules/ejs/lib/ejs.js:528:68)
    at Object.Template.compile (/usr/local/lib/node_modules/polymer-cli/node_modules/ejs/lib/ejs.js:444:12)
    at Object.compile (/usr/local/lib/node_modules/polymer-cli/node_modules/ejs/lib/ejs.js:289:16)
    at handleCache (/usr/local/lib/node_modules/polymer-cli/node_modules/ejs/lib/ejs.js:148:16)
    at Object.exports.render (/usr/local/lib/node_modules/polymer-cli/node_modules/ejs/lib/ejs.js:316:10)
    at copy.process (/usr/local/lib/node_modules/polymer-cli/node_modules/mem-fs-editor/lib/actions/copy-tpl.js:14:18)
    at applyProcessingFunc (/usr/local/lib/node_modules/polymer-cli/node_modules/mem-fs-editor/lib/actions/copy.js:13:16)
    at EditionInterface.exports._copySingle (/usr/local/lib/node_modules/polymer-cli/node_modules/mem-fs-editor/lib/actions/copy.js:51:16)
    at EditionInterface.<anonymous> (/usr/local/lib/node_modules/polymer-cli/node_modules/mem-fs-editor/lib/actions/copy.js:38:10)
    at Array.forEach (native)
Jerrys-MBP:test-element Tykin$ node -v
v5.7.0
Jerrys-MBP:test-element Tykin$ npm -v
3.6.0

```

**Comment by malthusyau** [comment]
Same issue for me on both mac (node v6.2.0 npm 3.8.9) and linux (node 5.10.1 npm 3.8.3)

**Comment by FredKSchott** [comment]
We've tracked this down to an error in one of yeoman's dependencies introduced within the last 24 hours:
https://github.com/SBoudrias/mem-fs-editor/issues/59
https://github.com/mde/ejs/issues/166

Until recently this `copyTpl()` could accept an array of file patterns to match against, even though this wasn't explicitly supported by it's dependency ejs. A recent change there now causes this to fail instead. We're updating our call to `copyTpl()` to use a single pattern string instead of the collection of patterns that previously worked.

PR incoming

**Comment by NicolasRannou** [comment]
+1
👍

**Comment by justinfagnani** [comment]
Fixed in #207

**Comment by FredKSchott** [comment]
Just pushed to npm as well
